lxd/loki/loki.go

```diff
@@ -273,7 +273,8 @@ func (c *Client) HandleEvent(event api.Event) {
 			if shared.StringInSlice(k, c.cfg.labels) {
 				_, ok := entry.labels[k]
 				if !ok {
-					entry.labels[k] = v
+					// Label names may not contain any hyphens.
+					entry.labels[strings.ReplaceAll(k, "-", "_")] = v
```

Comment:
That replacement makes sense, for `name` and `project`.

Since `loki.labels` allows a user to provide anything, maybe we should validate what is set matches the "valid label name regex": `[a-zA-Z_:][a-zA-Z0-9_:]*`.

Reply:
No, we don't need to validate this. `k` will always be something like `context-foo`. A user will need to add `context-foo` to `loki.labels`, so the exact match. But Loki itself will be sent the correctly formatted key.

Reply:
Indeed, thanks for the explanation!